src/participants/create_project.py

```diff
@@ -148,55 +153,80 @@ def handle_wi(self, wid):
+            # TODO: construct repos and build paths for a devel build
 
         if len(prj_parts) >= 3 and prj_parts[-3] == "feature":
-            link = ":".join(prj_parts[0:-3])
+            linked_project = ":".join(prj_parts[0:-3])
             fea = "%s#%s" % (prj_parts[-2], prj_parts[-1])
             # Go through each bugzilla we support
             for (bugzillaname, bugzilla) in self.bzs.iteritems():
                 for match in bugzilla['compiled_re'].finditer(fea):
                     try:
-                        summary = bugzilla['interface'].bug_get(match.group('key'), 0)['summary']
-                        desc = bugzilla['interface'].comment_get(match.group('key'), 0)['text']
+                        summary = bugzilla['interface'].bug_get(
+                            match.group('key'), 0)['summary']
+                        desc = bugzilla['interface'].comment_get(
+                            match.group('key'), 0)['text']
                     except BugzillaError, error:
                         if error.code == 101:
                             print "Bug %s not found" % bugnum
                         else:
                             raise
-            if project not in project_list: create = True
+            if project not in project_list:
+                create = True
 
-        if link and link in project_list:
-            links.append(link)
-            repolinks.update(self.get_repolinks(wid, link))
+        if linked_project and linked_project in project_list:
+            linked_projects.append(linked_project)
+            repolinks.update(self.get_repolinks(wid, linked_project))
 
         if create:
-            result = self.obs.createProject(project, repolinks, desc=desc, title=summary, mechanism=mechanism,
-                                            links=links, maintainers=maintainers, build=build, block=block)
+            if not repolinks:
+                # Creating a project with no repos makes no sense
+                # as we are only doing this to perform a test build
+                #
+                # It is debatable that we could want to create
+                # projects that have no repos to build against but
+                # there have been real-world issues where such
+                # projects are not properly detected and reported by
+                # the workflow.
+                #
+                # If this participant is developed to be a general
+                # purpose create_project then existing usage should be
+                # audited.
+                raise RuntimeError(
+                    "No suitable repos found in %s (must contain an arch in the name)"
+                    % project)
+
+            result = self.obs.createProject(
+                project, repolinks, desc=desc, title=summary, mechanism=mechanism,
+                                            links=linked_projects, maintainers=maintainers, build=build, block=block)
```

Comment:
Also here, I'd argue for 

            result = self.obs.createProject(
                project, repolinks, desc=desc, title=summary, mechanism=mechanism,
                links=linked_projects, maintainers=maintainers, build=build, block=block
            )